daemons/based/based_remote.c

```diff
@@ -450,10 +450,16 @@ cib_remote_msg(gpointer data)
     pcmk__client_t *client = ncd->new_client;
     xmlNode *command = NULL;
     int rc;
-    int timeout = 1000;
+    int timeout_sec = 1;
 
     if (pcmk_is_set(client->flags, pcmk__client_authenticated)) {
-        timeout = -1;
+        timeout_sec = 60;
+    }
+
+    if (time(NULL) >= ncd->start_time + timeout_sec) {
+        free(ncd);
+        crm_info("Error reading from remote client: %s", pcmk_rc_str(ETIME));
+        return -1;
```

Comment:
If start_time is 0, this is the start of a new message, so no need to check the timeout (as in cib_remote_callback_dispatch()), unless I'm missing something

Reply:
Updated to use the same block as the top of cib_remote_callback_dispatch.